### ISSUE TITLE
Adding months to date across DST does not preserve hour

### ISSUE THREAD

**Issue by JBaczuk** [moment/moment#5825]
**Describe the bug**
If you add months to a date across DST, it does not preserve the hour as stated in the docs:

> There are also special considerations to keep in mind when adding time that crosses over daylight saving time. If you are adding years, months, weeks, or days, the original hour will always match the added hour.

**To Reproduce**
```
> date = new Date()
2021-02-04T16:37:37.790Z
> moment(date).add(6, 'months').toISOString()
'2021-08-04T15:37:37.790Z'
```

**Expected behavior**
```
> date = new Date()
2021-02-04T16:37:37.790Z
> moment(date).add(6, 'months').toISOString()
'2021-08-04T16:37:37.790Z'
```
**Screenshots**

**Desktop (please complete the following information):**
 - OS: MacOS 11.2
 - Node: `v14.15.4`
 - Moment Version: `2.29.1`

**Moment-specific environment**

* The time zone setting of the machine the code is running on: `America/Denver`
* The time and date at which the code was run: `1612456862588`
* Other libraries in use (TypeScript, Immutable.js, etc): `n/a`

Please run the following code in your environment and include the output:
```
console.log((new Date()).toString())
console.log((new Date()).toLocaleString())
console.log((new Date()).getTimezoneOffset())
console.log(navigator.userAgent)
console.log(moment.version)
```

Result:
```
> console.log((new Date()).toString())
Thu Feb 04 2021 09:42:43 GMT-0700 (Mountain Standard Time)
> console.log((new Date()).toLocaleString())
2/4/2021, 9:42:43 AM
> console.log((new Date()).getTimezoneOffset())
420
> console.log(navigator.userAgent)
Uncaught ReferenceError: navigator is not defined
> console.log(moment.version)
2.29.1
```

**Comment by benzispawn** [comment]
I don't this issue on the 2.24.0
To Reproduce
`> date = new Date()
2021-02-17T15:37:07.718Z
> moment(date).add(6, 'months').toISOString()
'2021-08-17T15:37:07.718Z'`

Result: 
`> console.log((new Date()).toString())
Wed Feb 17 2021 12:35:22 GMT-0300 (Brasilia Standard Time)
undefined
> console.log((new Date()).toLocaleString())
e()).getTimezoneOffset())
console.log(navigator.us2/17/2021, 12:35:22 PM
undefined
> console.log((new Date()).getTimezoneOffset())
180
undefined
> console.log(navigator.userAgent)
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_6_7) AppleWebKit/534.36 (KHTML, like Gecko) NodeJS/v0.4.7 Chrome/13.0.767.0 Safari/534.36
undefined
> console.log(moment.version)
2.24.0`

**Comment by benzispawn** [comment]
I don't see the problem on 2.29.1 either
`> date = new Date()
2021-02-17T15:39:26.660Z
> moment(date).add(6, 'months').toISOString()
'2021-08-17T15:39:26.660Z'
> console.log((new Date()).toString())
Wed Feb 17 2021 12:39:46 GMT-0300 (Brasilia Standard Time)
undefined
> console.log((new Date()).toLocaleString())
2/17/2021, 12:39:46 PM
undefined
> console.log((new Date()).getTimezoneOffset())
180
undefined
> console.log(navigator.userAgent)
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_6_7) AppleWebKit/534.36 (KHTML, like Gecko) NodeJS/v0.4.7 Chrome/13.0.767.0 Safari/534.36
undefined
> console.log(moment.version)
2.29.1`

**Comment by JBaczuk** [comment]
@benzispawn I think you are running it in your local time (Brasilia Standard Time) which does not observe daylight savings, correct? Therefore, you most likely wouldn't see this issue.

The important detail is that it was run in `America/Denver` timezone which does currently observe daylight savings.

**Comment by JBaczuk** [comment]
@marwahaha Why was this closed?

**Comment by jsimpson** [comment]
We are currently experiencing the same problem.

**Comment by adamni21** [comment]
@JBaczuk for me the behavior is as described by the docs here: [link](https://momentjs.com/docs/#/manipulating/add/:~:text=There%20are%20also%20special%20considerations%20to%20keep%20in%20mind%20when%20adding%20time%20that%20crosses%20over%20daylight%20saving%20time.%20If%20you%20are%20adding%20years%2C%20months%2C%20weeks%2C%20or%20days%2C%20the%20original%20hour%20will%20always%20match%20the%20added%20hour.)
That is the local time of day for both moments stays the same, however it'll shift the UTC time of day by one hour.

The following code produces the output below (node: 20.17.0; moment: 2.30.1;):
```javascript
import moment from "moment";

const date = moment({ date: 25, month: 1, year: 2025, hour: 10 });
const plusOneMonth = moment(date).add(1, "month");

const timeZone = Intl.DateTimeFormat().resolvedOptions().timeZone;

console.log(date.format("HH:mm Do MMM YYYY ZZ") + " " + timeZone);
console.log(plusOneMonth.format("HH:mm Do MMM YYYY ZZ") + " " + timeZone);
console.log(date.utc().format("HH:mm Do MMM YYYY ZZ"));
console.log(plusOneMonth.utc().format("HH:mm Do MMM YYYY ZZ"));
```
output:

```
10:00 25th Feb 2025 -0700 America/Denver
10:00 25th Mar 2025 -0600 America/Denver
17:00 25th Feb 2025 +0000
16:00 25th Mar 2025 +0000
```

**Comment by mattjohnsonpint** [comment]
Agreed. The documented guarantee refers to the local hour, not the UTC hour shown by `toISOString()`. Across DST, local `09:37` remains `09:37` while the offset changes, so the ISO UTC hour correctly shifts. I'm closing this as expected behavior.
